src/Compilers/CSharp/Test/Semantic/Semantics/RangeTests.cs

```diff
@@ -0,0 +1,415 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Xunit;
+
+namespace Microsoft.CodeAnalysis.CSharp.UnitTests
+{
+    public class RangeTests : CompilingTestBase
+    {
+        private static readonly string RangeStruct = @"
+using System.Collections;
+using System.Collections.Generic;
+namespace System
+{
+    public struct Range : IEnumerable<int>
+    {
+        public int Start { get; }
```

Comment:
>Start [](start = 19, length = 5)

Given that this is a closed (inclusive) range, probably want to call the fields First and Last.

Reply:
Also possibly suggest `public readonly` fields instead of properties.

---
In reply to: [152159918](https://github.com/dotnet/roslyn/pull/23312#discussion_r152159918) [](ancestors = 152159918)

Reply:
This doesn't really matter. LDM hasn't decided, and fields vs. properties don't matter either (these are not anywhere close to the type that will go into the BCL, nor are they intended to be)

Reply:
I think it should be `readonly struct` as well as having `readonly` fields. Even this is not the most final form, given it's the first, it's fairly likely to be quickly copied by others. 😄